doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,352 @@
+            |                                                          |                |
+            v                                                          |                |
+       +---------+---------+---------+                                 |                |
+       | Bucket1 | ...     | BucketN |                                 v                |
+       +=========+=========+=========+                          +---------------+       |
+       | Header  | ...     | ...     |                          | OID           |<---------------+
+       +---------+---------+---------+                          | Num Chunk     |       |        |
+    +--| Entry0  | ...     | ...     |                          | <Offset, len> |       |        |
+    |  +---------+---------+---------+                          | <Offset, len> |       |        |
+    |  | ...     | ...     | ...     |                          | ...           |       |        |
+    |  +---------+---------+---------+                          +---------------+       |        |
+    |  | EntryK  | ...     | ...     |                                                  |        |
+    |  +---------+---------+---------+                                                  |        |
+    |                                                                                   |        |
+    |                                                                                   |        |
+    |  +---------------+  +----------+  +----------+                      segment       v        |
+    +->| object_info_t |  | omap     |  | data     |               +------------+------------+   |
+       +===============+  +==========+  +==========+          +--->| Block0     | Block1     |   |
+       | CID, OID      |  | CID, OID |  | CID, OID |          |    +============+============+   |
+       | Type          |  | Type     |  | Type     |          |    | Header     |            |   |
+       | Offset        |  | Offset   |  | Offset   |          |    +------------+            +   |
+       | Segment       |  | Segment  |  | Segment  |----------+    | Small data | Data       |   |
+       +---------------+  +----------+  +----------+               +------------+            +   |
+                                                                +--| OID meta Entries|       |   |
+                                                                |  +------------+------------+   |
+                                                                |                                |
+                                                                +--------------------------------+
+
+
+WAL
+---
+ToDo: need discussion before detailed design
+
+
+Partition
+---------
+Initially, PoseidonStore employs static allocation of partition. The number of sharded partitions
+is fixed and the size of each partition also should be configured before running cluster.
+
+
+Cache
+-----
+ToDo: need discussion before detailed design
```

Comment:
I'd expect that as you implement this, the actual implementation will change dramatically.  I'm more interested in generally how you structure the cache w.r.t. how other components interact with it and how lifetimes are managed.

Reply:
See below